javascript/ql/experimental/adaptivethreatmodeling/lib/experimental/adaptivethreatmodeling/EndpointFeatures.qll

```diff
@@ -184,14 +184,26 @@ module NeighborhoodBodies {
    * in the subtree.
    */
   Raw::AstNode getNeighborhoodAstNode(Raw::AstNode node) {
-    if getNumDescendents(node.getParentNode()) > maxNumDescendants()
+    if
+      // `node` will always have a parent as we start at and endpoint
+      node.getParentNode() = getOutermostEnclosingFunction(node) or
+      getNumDescendents(node.getParentNode()) > maxNumDescendants()
     then result = node
     else result = getNeighborhoodAstNode(node.getParentNode())
   }
 
   /** Count number of descendants of an AST node */
   int getNumDescendents(Raw::AstNode node) { result = count(node.getAChildNode*()) }
 
+  private ASTNode getContainer(ASTNode node) {
```

Comment:
OOI, what's the difference between a container and a parent?

Reply:
Containers skip up to an enclosing function, parents step one level up the AST graph (e.g. for 
```
f() { 
  if(endpoint) {
    … 
  } 
}
```
the container is `f()` but the parent is `if(…)`

Reply:
But then why do we need `getContainer*`?

Reply:
If an endpoint is enclosed in a function that's enclosed in a function, do the `enclosingFunction` features look at the outermost function?